### DISCUSSION TITLE
Support for ElasticSearch 6.3.x Rollups

### DISCUSSION THREAD

**Discussion by themanwithanrx7** [grafana/grafana#56258] [Observability]
Elasticsearch has finally added the ability to do automated data rollup of indexes.  They have also provided a unified search endpoint so both "live" data and rolled-up data can be searched and merged results returned.

https://www.elastic.co/blog/elasticsearch-6-3-0-released

https://www.elastic.co/guide/en/elasticsearch/reference/current/_rollup_uses_standard_query_dsl.html

It would be nice if we could add support for this to Grafana so displaying long-term data is possible.  Since the rolled up data is stored in a seperate index we could most likely just add another index attribute/pattern so the query knows which index/rollup to search from.

**Comment by pyro2927** [comment]
Might be even easier to just target the API endpoint `_rollup_search`, as that's [supported on both live and rollup indexes](https://www.elastic.co/guide/en/elasticsearch/reference/6.x/rollup-search.html#_searching_both_historical_rollup_and_non_rollup_data).  Then the Datasource config could just be set to point at `live_index,rollup_index`.

**Comment by niempy** [comment]
+1

**Comment by vvvprabhakar** [comment]
+1

**Reply by markniemeijer** [reply]
+1

**Comment by markniemeijer** [comment]
+1

**Comment by dragosrosculete** [comment]
+100

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82944. Feel free to continue the discussion around this there. Thank you!
